### PR TITLE
Use optional chaining instead of get from lodash

### PR DESCRIPTION
**Pull request by perunt**
Fixes RNBW-3339

## What changed (plus any additional context for devs)
Optional chaining is used instead of the lodash get method. Updated the type for nativeCurrency props

## Dev checklist for QA: what to test
it's basically used almost everywhere

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Review by osdnk — CHANGES_REQUESTED** [review]
- I didn't test it yet!

The major problem I observed is using `|| defaultValue`.
This is problematic, because 


```js
const defaultValue = 20;
const obj = {
   value: 0
}
```

Previously, 
```js
get(obj, 'value', defautlValue) === 0
```

With your changes (the approach you use consecutively: 
```js
(obj?.value || defaultValue) === 20
```
because `0` is false-ish (so does empty string e.g.).

This is at least one problem I am aware of and together with introducing optional chaining, we got a tool for this issue:
```js
(obj?.value ?? defaultValue) === 0
```

This looks better and is the exact answer for the problem lodash was about to solve by introducing `get` 

Please, use this pattern

File: src/parsers/gas.ts [hunk 1]
```diff
@@ -302,7 +306,7 @@ export const parseGasFeesBySpeed = (
   baseFeePerGas: GasFeeParam,
   gasLimit: BigNumberish,
   priceUnit: BigNumberish,
-  nativeCurrency: string
+  nativeCurrency: keyof typeof supportedNativeCurrencies
```
Comment (osdnk) [inline src/parsers/gas.ts:309]: @markdalgleish  can you if this is hot or not?

**Comment by perunt** [comment]
@osdnk, thanks for the comments, yeah, it's better to use nullish coalescing. I updated. Have you any other comments on that?

**Review by brunobar79 — COMMENTED** [review]
Good stuff!

Added some minor comments. Ping me after those are fixed and CI is passing and I'll approve.

File: src/components/exchange/CurrencySelectionList.js [hunk 2]
```diff
@@ -45,7 +44,7 @@ const CurrencySelectionList = (
   ref
 ) => {
   const skeletonTransitionRef = useRef();
-  const noResults = get(listItems, '[0].data', []).length === 0;
+  const noResults = (listItems?.[0]?.data ?? []).length === 0;
```
Comment (osdnk) [inline src/components/exchange/CurrencySelectionList.js:47]:
```suggestion
  const noResults = listItems?.[0]?.data?.length ?? 0 === 0;
```

or

```suggestion
  const noResults = !!listItems?.[0]?.data?.length
```

File: src/helpers/buildWalletSections.js [hunk 3]
```diff
@@ -439,9 +429,9 @@ const sortImagesToPreload = images => {
   const filtered = compact(flattenDeep(images));
   const grouped = groupBy(filtered, property('priority'));
   return [
-    ...get(grouped, 'high', []),
-    ...get(grouped, 'normal', []),
-    ...get(grouped, 'low', []),
+    ...(grouped?.high ?? []),
+    ...(grouped?.normal ?? []),
+    ...(grouped?.low ?? []),
```
Comment (osdnk) [inline src/helpers/buildWalletSections.js:434]:
```suggestion
    ...grouped?.high,
    ...grouped?.normal,
    ...grouped?.low,
```

I think this is the same

**Review by osdnk — APPROVED** [review]

**Comment by osdnk** [comment]
conflicts

File: src/components/exchange/CurrencySelectionList.js (hunk 2 above)
Comment (perunt) [reply src/components/exchange/CurrencySelectionList.js:47]: Good point, thanks. Will use with one NOT for the same result
`const noResults = !listItems?.[0]?.data?.length`

File: src/helpers/buildWalletSections.js (hunk 3 above)
Comment (perunt) [reply src/helpers/buildWalletSections.js:434]: yea, just Prettier argued about it

**Review by brunobar79 — APPROVED** [review]
🚀 🚀 🚀

**Review by tchayen — APPROVED** [review]
I read through the code and ran on Android and iOS. Works fine 👌.

File: src/helpers/buildWalletSections.js (hunk 3 above)
Comment (terrysahaidak) [reply src/helpers/buildWalletSections.js:434]: Since it passes the CI i would leave it as is

File: src/parsers/poap.js [hunk 4]
```diff
@@ -8,7 +7,7 @@ import { imageToPng } from '@rainbow-me/handlers/imgix';
  */
 
 export const parsePoaps = data => {
-  const poaps = get(data, 'data', null);
+  const poaps = data?.data ?? null;
```
Comment (terrysahaidak) [inline src/parsers/poap.js:10]: is it ok for this function to throw? i mean on the very next line it assumes that poaps is an array

**Review by terrysahaidak — APPROVED** [review]
LGTM except for a single comment with a potential bug not related to this PR but still you touched it

File: src/parsers/poap.js (hunk 4 above)
Comment (perunt) [reply src/parsers/poap.js:10]: It's handled in upper try-catch. For now, it's not broken anything, but it should be changed definitely

**Review by jkadamczyk — APPROVED** [review]
Looks good to me!

**Review by skylarbarrera — APPROVED** [review]
🙌

**Review by estebanmino — APPROVED** [review]

**Pull request merged**